features/telemetry/listeners/src/main/java/org/opennms/netmgt/telemetry/listeners/factory/TcpListenerFactory.java

```diff
@@ -68,9 +66,6 @@ public Listener createBean(ListenerDefinition listenerDefinition) {
         if (parser.size() != listenerDefinition.getParsers().size()) {
             throw new IllegalArgumentException("Each parser must be of type TcpParser but was not.");
         }
-        final TcpListener listener = new TcpListener(listenerDefinition.getName(), parser.iterator().next(), telemetryRegistry.getMetricRegistry());
-        final BeanWrapper wrapper = PropertyAccessorFactory.forBeanPropertyAccess(listener);
```

Comment:
I believe this was done within the factory itself rather that outside because of reflection issues with the OSGi proxy classes.

Does this work differently now?

Reply:
I don't know if this has been the case before. But I verified that it works on standalone OpenNMS and on Minion.

The instances returned from the factories are not proxied (whereas the factories itself are). Therefore the `BeanWrapper` should work just fine.

Reply:
Sounds good. Thanks for the confirmation.